test/unit/puppetlabs/services/jruby_pool_manager/jruby_interpreter_test.clj

```diff
@@ -8,15 +8,26 @@
     (let [jruby-interpreter (jruby-internal/create-scripting-container
                               (jruby-testutils/jruby-config))
           jruby-env (.runScriptlet jruby-interpreter "ENV")]
-      ; $HOME and $PATH are left in by `jruby-env`
+      ;; $HOME and $PATH are left in by `jruby-env`
+      ;; Note that other environment variables are allowed through (e.g. `HTTP_PROXY` -
+      ;; see jruby-core/env-allowed-list for the full list), but are not
+      ;; expected to show up in most environments.
+      ;; If this test starts failing because CI systems or local development
+      ;; environments do have these variables set, we may need to revisit
+      ;; this test.
       (is (= #{"HOME" "PATH" "GEM_HOME" "JARS_NO_REQUIRE" "JARS_REQUIRE" "RUBY"}
```

Comment:
I made a similar comment in the corresponding [puppetserver PR](https://github.com/puppetlabs/puppetserver/pull/1289#discussion_r88682219).  I think it might be good here to filter out the http / no proxy variables that might end up being passed through to JRuby now (if a user happens to have had them set on their system when running the tests) before doing the equality comparison.  That way, this and the related `jruby-configured-env-vars` test could pass either way.